packages/hdwallet-provider/src/constructor/getOptions.ts

```diff
@@ -20,29 +17,31 @@ const isPrivateKeys = (
 const isPrivateKey = (
   credentials: LegacyConstructor.Credentials
 ): credentials is PrivateKey =>
-  !isPrivateKeys(credentials) && !isMnemonicPhrase(credentials);
+  typeof credentials === "string" &&
+  credentials.length === 64 &&
+  // this is added since parseInt(mnemonic) should equal NaN and private keys
+  // should parse into a valid number
+  parseInt(credentials) !== NaN;
```

Comment:
This should be `parseInt(credentials, 16)`, right?  Without the 16, it won't work unless there's a `0x` in front, and I don't think we want to require that.

I also think using `parseInt` in this way is pretty iffy, because the number is 32 bytes long.  It technically works, but I think if you want to do this it's at least worth making a comment saying something like "yes, this works even if the number is outside the safe integer range!".

Reply:
Yup, the 16 is definitely missing! Mnemonics technically can parse into a number if they start with the letters a-f so @cds-amal recommended checking for " "s as well. I'll add a comment here mentioning that "f" *64 still parses into a number in JS.